servers/quarkus-common/src/main/java/org/projectnessie/quarkus/config/datasource/DataSourceActivator.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.config.datasource;
+
+import io.smallrye.config.ConfigSourceInterceptor;
+import io.smallrye.config.ConfigSourceInterceptorContext;
+import io.smallrye.config.ConfigValue;
+import org.projectnessie.quarkus.config.VersionStoreConfig.VersionStoreType;
+
+/**
+ * Activates a data source based on the current Nessie configuration under {@code
+ * nessie.version.store.persist.jdbc.datasource}, and deactivates all other data sources.
+ *
+ * <p>If the version store type is not JDBC, all data sources are deactivated.
+ */
+public class DataSourceActivator implements ConfigSourceInterceptor {
+
+  private static final int APPLICATION_PROPERTIES_CLASSPATH_ORDINAL = 250;
+
+  @Override
+  public ConfigValue getValue(ConfigSourceInterceptorContext context, String name) {
+    if (name.startsWith("quarkus.datasource.") && name.endsWith(".active")) {
+      boolean active = false;
+      if (versionStoreType(context) == VersionStoreType.JDBC) {
+        active = dataSourceName(name).equals(activeDataSourceName(context));
+      }
+      return newConfigValue(active);
```

Comment:
Wonder whether we always want a new `ConfigValue` - even for the case when the value isn't changed.

WDYT about this?
```java
  @Override
  public ConfigValue getValue(ConfigSourceInterceptorContext context, String name) {
    ConfigValue configValue = context.proceed(name);
    if (name.startsWith("quarkus.datasource.") && name.endsWith(".active")) {
      boolean active = false;
      if (versionStoreType(context) == VersionStoreType.JDBC) {
        active = dataSourceName(name).equals(activeDataSourceName(context));
      }
      if (active != Boolean.parseBoolean(configValue.getValue())) {
        configValue = ConfigValue.builder()
          .withValue(active ? "true" : "false")
          .withConfigSourceOrdinal(configValue.getSourceOrdinal() + 1).build();
      }
    }
    return configValue;
  }
```